txcache/selectionTracker.go

```diff
@@ -421,3 +421,29 @@ func (st *selectionTracker) getChainOfTrackedPendingBlocks(
 	slices.Reverse(chain)
 	return chain, nil
 }
+
+func (st *selectionTracker) getVirtualNonceOfAccountWithRootHash(
+	address []byte,
+	blockchainInfo common.BlockchainInfo,
+) (uint64, []byte, error) {
+	latestCommitedBlockHash := blockchainInfo.GetLatestCommittedBlockHash()
+	if latestCommitedBlockHash == nil {
+		return 0, nil, errNilLatestCommitedBlockHash
+	}
+
+	latestCommitedBlock, ok := st.blocks[string(latestCommitedBlockHash)]
```

Comment:
Oh, sorry, now I realize how you did it, good. Ignore my comment about "the cache should return the hash given its view".